Add use cases and demo to Shared Storage explainer (#2976)
GoogleChrome/developer.chrome.com: site/en/docs/privacy-sandbox/shared-storage/index.md

The Shared Storage API intends to support many use cases, replacing several
existing uses for third-party cookies. This may include:

> *  **Frequency capping** - When the same ad is shown to the user repeatedly, the ad starts to lose its effectiveness. Frequency capping allows you to limit the number of times an ad can be shown to a user. The count of how many times the user has seen the ad is saved in shared storage, and can be accessed [cross-site](https://web.dev/same-site-same-origin/). 
> *  **A/B testing, such as brand lift experiment** - To see if an experiment has the desired effect, you can conduct A/B testing across multiple sites. As an advertiser, you can choose to render a different ad based on what group the user is assigned to. The group assignment is saved in shared storage.
> * **Creative rotation** - An advertiser may want to apply different strategies to an ad campaign, and rotate the creatives to increase effectiveness of the ads. Shared storage can be used to run different rotation strategies, such as sequential rotation and evenly-distributed rotation, across different sites.
> * **Known customer** - You might want to render a different element based on whether the user has been seen before at a different site. For example, a payment provider may want to render a "Register" or "Buy now" button based on whether the user has registered before at the payment provider's site. Shared storage can be used to set the status of the user.
> * **Reach measurement and custom reporting** - Advertisers can save the user information, such as demographics, reach, frequency measurement, and conversion measurement in shared storage. Then [Private Aggregation API](https://github.com/alexmturner/private-aggregation-api) can be used transmit and processes the data in a privact-preserving way to generate an aggregated report.

Reviewer: ```suggestion
*  **Frequency capping** - When the same ad is shown repeatedly to a user, the ad starts to lose its effectiveness. Frequency capping allows you to limit the number of times an ad can be shown to a user. The count of how many times the user has seen the ad is saved in shared storage and can be accessed [cross-site](https://web.dev/same-site-same-origin/). 
*  **A/B testing, such as brand lift experiment** - To see if an experiment has the desired effect, you can conduct A/B testing across multiple sites. As an advertiser, you can choose to render a different ad based on what group the user is assigned to. The group assignment is saved in shared storage.
* **Creative rotation** - An advertiser may want to apply different strategies to an ad campaign, and rotate the creatives to increase effectiveness of the ads. Shared storage can be used to run different rotation strategies, such as sequential rotation and evenly-distributed rotation, across different sites.
* **Known customer** - You may want to render a different element based on whether the user was seen on a different site. For example, a payment provider may want to render a "Register" or "Buy now" button based on whether the user has registered at the payment provider's site. Shared storage can be used to set the user's status.
* **Reach measurement and custom reporting** - Advertisers can save user information, such as demographics, reach, frequency measurement, and conversion measurement, in shared storage. The [Private Aggregation API](https://github.com/alexmturner/private-aggregation-api) can be used transmit and processes the data in a privacy-preserving way to generate an aggregated report.
```
